### PR TITLE
Add unlocked_addresses, alias_address, nft_address and AliasTransition;

### PR DESCRIPTION
**Pull request by Thoralf-M**
Change max length for TaggedDataPayload

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Test

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: types/src/block/error.rs [hunk 1]
```diff
@@ -276,6 +277,7 @@ impl fmt::Display for Error {
                 write!(f, "milestone signatures are not unique and/or sorted")
             }
             Error::MissingAddressUnlockCondition => write!(f, "missing address unlock condition"),
+            Error::MissingAliasTransitionType => write!(f, "missing alias state transition type"),
```
Comment (Thoralf-M) [inline types/src/block/error.rs]: ### 
```suggestion
            Error::MissingAliasTransitionType => write!(f, "missing alias transition type"),
```

Comment (Thoralf-M) [inline types/src/block/error.rs]: ### 
```suggestion
            Error::MissingAliasTransitionType => write!(f, "missing alias transition type"),
```

File: types/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- `alias_address()` to `AliasOutput`;
+- `nft_address()` to `NftOutput`;
+- `unlocked_addresses()` to `Output`;
+- `MissingAliasTransitionType` error variant;
+
+### Changed
+
+- Reduced `DATA_LENGTH_RANGE` of `TaggedDataPayload`;
+
 First release based on `bee-api-types` and `bee-block`.
```
Comment (thibault-martinez) [inline types/CHANGELOG.md]: Can you put that before ### Added please ?

File: types/src/block/output/mod.rs [hunk 3]
```diff
@@ -140,6 +140,48 @@ impl Output {
         }
     }
 
+    /// Returns the the address that is required to unlock this [`Output`] and the alias or nft address that gets
```
Comment (thibault-martinez) [inline types/src/block/output/mod.rs]:
```suggestion
    /// Returns the address that is required to unlock this [`Output`] and the alias or nft address that gets
```

File: types/src/block/output/mod.rs [hunk 4]
```diff
@@ -140,6 +140,48 @@ impl Output {
...
+    /// unlocked by it, if its an alias or nft.
+    pub fn unlocked_addresses(
```
Comment (thibault-martinez) [inline types/src/block/output/mod.rs]: Can you put it after the regular getters please ?

File: types/CHANGELOG.md [hunk 5]
```diff
@@ -21,7 +21,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## 1.0.0-rc.1 - 2022-XX-XX
 
+### Added
+
 First release based on `bee-api-types` and `bee-block`.
```
Comment (thibault-martinez) [inline types/CHANGELOG.md]: Before :trollface:

File: types/src/block/output/mod.rs [hunk 6]
```diff
@@ -184,6 +184,48 @@ impl Output {
         }
     }
 
+    /// Returns the address that is required to unlock this [`Output`] and the alias or nft address that gets
+    /// unlocked by it, if its an alias or nft.
+    pub fn unlocked_addresses(
+        &self,
+        current_time: u32,
+        alias_transition: Option<AliasTransition>,
+        output_id: OutputId,
```
Comment (thibault-martinez) [inline types/src/block/output/mod.rs:192]: Do you mind swapping these 2 params ? Conventionally we'd put the options at the end

File: types/src/block/output/mod.rs [hunk 7]
```diff
@@ -184,6 +184,48 @@ impl Output {
         }
     }
 
+    /// Returns the address that is required to unlock this [`Output`] and the alias or nft address that gets
+    /// unlocked by it, if its an alias or nft.
+    pub fn unlocked_addresses(
```
Comment (thibault-martinez) [inline types/src/block/output/mod.rs]: (trying to think of a better name)

File: types/src/block/output/mod.rs [hunk 8]
```diff
@@ -184,6 +184,48 @@ impl Output {
         }
     }
 
+    /// Returns the address that is required to unlock this [`Output`] and the alias or nft address that gets
+    /// unlocked by it, if its an alias or nft.
```
Comment (thibault-martinez) [inline types/src/block/output/mod.rs]:
```suggestion
    /// unlocked by it, if it's an alias or nft.
```

File: types/src/block/output/mod.rs (hunk 7 above)
Comment (Thoralf-M) [reply types/src/block/output/mod.rs]: Maybe `required_and_unlocked_address()`?

Comment (thibault-martinez) [reply types/src/block/output/mod.rs]: Sounds good to me

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**